tests/test_client.py

```diff
@@ -397,7 +397,7 @@ def test_path_is_not_double_encoded(self):
         self.assertEqual(req.path, "/get/test%20case")
 
         req = ClientRequest('get', "http://0.0.0.0/get/test%2fcase")
-        self.assertEqual(req.path, "/get/test%2fcase")
+        self.assertEqual(req.path, "/get/test/case")
```

Comment:
I think you break something here, because `test/case != test%2fcase` - these are two different resources.